dotnet-etcd/grpc/Rpc.grpc.cs

```diff
@@ -140,127 +140,128 @@ public static partial class RpcReflection {
             "MhcuZXRjZHNlcnZlcnBiLkFsYXJtVHlwZSJoCg1BbGFybVJlc3BvbnNlEiwK",
```

Comment:
Which version of `Grpc.Tools` was used to generate the stub ?. Also, Please ensure that the bundled `protoc` binary inside `Grpc.Tools` is used for code generation.

Reply:
Nevermind, I am in the process of updating packages. Will do this myself. Thanks for contributing.

Reply:
I did use `Grpc-Tools` version `1.22.0` with command `%UserProfile%\.nuget\packages\grpc.tools\1.22.0\tools\windows_x64\protoc.exe --proto_path=proto/ --grpc_out=. --csharp_out=. --csharp_opt=file_extension=.grpc.cs rpc.proto --plugin=protoc-gen-grpc=%UserProfile%\.nuget\packages\grpc.tools\1.22.0\tools\windows_x64\grpc_csharp_plugin.exe`

I also tried version `2.23.0` but got the same `descriptorData` blob. 

I'm new to generating protobuf stubs. I'm I doing it right @shubhamranjan?

Reply:
Thx

Reply:
1. I use the `protoc` binary found under `Grpc.Tools.X.X.X\tools\windows_x64` to generate stub.
2.  Here's a sample command that I run to generate file.  I am using v2.23.0 now in the latest alpha release.
```
protoc --proto_path=./grpc/proto --grpc_out grpc --csharp_out=./grpc/ --csharp_opt=file_extension=.grpc.cs rpc.proto --plugin=protoc-gen-grpc=grpc/packages/Grpc.Tools.2.23.0/tools/windows_x64/grpc_csharp_plugin.exe
```

Reply:
@shubhamranjan In that case we did the same. Did you get the same blob?

Reply:
I regenerated with 2.23.0. Hence, got a different blob.

Reply:
Tanks for fast response pushing out a new Nuget package

Reply:
No Probs. Also, I would love if you could test out the new package with single node and multi node cluster and tell me of any issues. Would help me fasten out the stable package.

Reply:
I tested `3.1.0-alpha` with local single node and local three node cluster, and it seams to work as excpeted. 

The new properties on `StatusResponse` all have values except for Error, probably since it's hard to trigger no disk error. 

When I kill one node nothing changes in `StatusResponse` except the expected incremented `raftIndex` and maybe `leader`.

When I kill two nodes and query the last one standing, an exception is thrown with `Status(StatusCode=Unknown, Detail="etcdserver: request timed out"`. The node logs says `rafthttp: health check for peer [X] could not connect: dial tcp [IP]:[port]: getsockopt: connection refused`. 
I guess an exception and not a Error in `StatusResponse` is expeceted?

Reply:
Hmm, I will check. Looks like something to do with the retry logic. Failover logic needs to be checked. Thanks for the feedback.

Reply:
Is an exception at `Status()` the expected outcome if 2/3 nodes are down?

Reply:
Nope. That's probably the second node was the one to which the first connection was made. As of now connection logic is random of healthy nodes.

Reply:
@maganvill The latest beta should work properly. Give it a try. Try with a cluster without auth enabled. I need to update docs for authentication enabled clusters before stable release.